pyiron_base/interfaces/has_dict.py

```diff
@@ -33,11 +37,50 @@
 __date__ = "Dec 20, 2023"
 
 
+def create_from_dict(obj_dict):
+    """
+    Create and restores an object previously written as a dictionary.
+
+    Args:
+        obj_dict (dict): must be the output of HasDict.to_dict()
+
+    Returns:
+        object: restored object
+    """
+    if "TYPE" not in obj_dict:
+        raise ValueError(
+            "invalid obj_dict! must contain type information and be the output of HasDict.to_dict!"
+        )
+    type_field = obj_dict["TYPE"]
+    module_path, class_name = _extract_module_class_name(type_field)
+    class_object = _import_class(module_path, class_name)
+    version = obj_dict.get("VERSION", None)
+    obj = class_object.instantiate(obj_dict, version)
+    obj.from_dict(obj_dict, version)
+    return obj
+
+
 class HasDict(ABC):
     __dict_version__ = "0.1.0"
 
-    @abstractmethod
+    @classmethod
+    def instantiate(cls, obj_dict: dict, version: str = None) -> "Self":
+        return cls()
+
     def from_dict(self, obj_dict: dict, version: str = None):
+        def load(inner_dict):
+            if not isinstance(inner_dict, dict):
+                return inner_dict
+            if not all(
+                k in inner_dict for k in ("NAME", "TYPE", "OBJECT", "DICT_VERSION")
+            ):
+                return {k: load(v) for k, v in inner_dict.items()}
+            return create_from_dict(inner_dict)
```

Comment:
Ok, I understand that the conversion is handled here, still I am wondering if this is what we want. I thought the intention was to get towards using `__getstate__()` and `__setstate__()`. Now with the executable and the server object using datacalsses for data storage internally, I was hoping that we could use a simple dictionary representation there where the class is stored by storing the attached dataclass.